src/pkg/packager/composer/list.go

```diff
@@ -22,17 +22,44 @@ import (
 type Node struct {
 	types.ZarfComponent
 
+	index int
+
 	vars   []types.ZarfPackageVariable
 	consts []types.ZarfPackageConstant
 
-	relativeToHead string
+	relativeToHeadOrURL string
```

Comment:
I think it could be good to remember the OCI url in a separate value just because a filepath.Join (the most common operation run on this value) would throw up if someone tried to use it with the URL.

Reply:
We could also have a simple helper function that gets it for you

Reply:
That makes sense, ended up going that route